File: Python/kyu-5/_0008__maximum-subarray-sum/solution.py
def max_sequence(arr):
    sum = total = 0
    for num in arr:
        sum += num
        sum = sum if sum > 0 else 0
        total = max(total, sum)
    return total

print(max_sequence([-2, 1, -3, 4, -1, 2, 1, -5, 4]));

File: JavaScript/kyu-5/_0008__maximum-subarray-sum/solution.js
const maxSequence = (arr) => {
  let sum = 0;
  let max = 0;
  arr.forEach((int) => {
    sum += int;
    sum < 0 && (sum = 0);
    sum > max && (max = sum);
  });
  return max;
};

console.log(maxSequence([-2, 1, -3, 4, -1, 2, 1, -5, 4]));
